api/backup/v1/artifacts.proto

```diff
@@ -85,20 +82,19 @@ message ListPitrTimerangesResponse {
 // Artifacts service provides public methods for managing backup artifacts.
 service ArtifactsService {
   // ListArtifacts returns a list of all backup artifacts.
-  rpc ListArtifacts(ListArtifactsRequest) returns (ListArtifactsResponse) {
+  rpc ListArtifacts(google.protobuf.Empty) returns (ListArtifactsResponse) {
```

Comment:
Same story, there will be no option to add filters parameters to this method. We will be forced to create new endpoint

Reply:
@artemgavrilov @maxkondr I wonder how do folks who promote the use of `google.protobuf.Empty` get away in such scenarios?

Doesn't that qualify for the next version of API, say `v2`?

Reply:
Filters should be optional, so adding them doesn't break backward compatibility. So it has not much sense to create v2 for that.

Reply:
For me it will be simpler to use something one, otherwise it became a mess over time.

Reply:
@ademidoff 
could you please check what will happen in case:
1.
```
rpc ListArtifacts(google.protobuf.Empty) returns (ListArtifactsResponse) {
  option (google.api.http) = {
       post: "/v1/backup/Artifacts/List"
  };
}
```
and call endpoint as:
```
curl -XPOST http://<domain>/v1/backup/Artifacts/List?foo=bar
```

2. that change API definition to:
```
rpc ListArtifacts(ListArtifactsRequest) returns (ListArtifactsResponse) {
  option (google.api.http) = {
       post: "/v1/backup/Artifacts/List"
  };
}
```
and call endpoint as:
```
curl -XPOST http://<domain>/v1/backup/Artifacts/List?foo=bar
```

Will it break smh?

Reply:
If your question is `How to pass URL parameters to a gRPC POST request`, then we have follow a special syntax for both examples. This `limitation` comes from the `http.proto` [doc](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L176-L180).

```
message ListArtifactsRequest {
  string foo = 5;
}

rpc ListArtifacts(ListArtifactsRequest) returns (ListArtifactsResponse) {
  option (google.api.http) = {
       post: "/v1/backup/Artifacts/List"
       body: "listartifactrequest" # Note the absence of "*"
  };
}

```

Reply:
my Q was: if we not utilize `google.protobuf.Empty` as request body, will it break smth in case later we need to change it to custom message in order to support list parameters?

Reply:
My understanding is it won't break anything. We can replace `.Empty` with with a different proto message when we need to.

Reply:
It will break gRPC API, which is public too

Reply:
Changes reverted, `gogle.protob.Empty` is not used anymore.